### DISCUSSION TITLE
DBT state weird behavior

### DISCUSSION THREAD

**Discussion by benbenbang** [dbt-labs/dbt-core#7264] [Q&A]
Hello all,

- Context: I am currently helping our BI colleagues to build a slim CI. To do so, I would like to leverage the `state:modifed` flag to speed up the (Self-host Gitlab) CI tests and make it more relevant.
- The `manifest.json` is stored by `git lfs`
- I use exactly the same user, db, warehouse to run the tests
- `DBT_ARTIFACT_STATE_PATH` is properly set
- DBT Version: 1.3.2

Details:
```
Core:
  - installed: 1.3.2
  - latest:    1.4.5 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - snowflake: 1.3.0 - Update available!

  At least one plugin is out of date or incompatible with dbt-core.
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation
```

Issue:
Firstly, I am not super savvy about DBT, I am learning by building and reading the docs (the docs are actually quite nice)
I am able to run 
- `dbt ls -s state:modified`
- `dbt run --models state:modified`
and see only 5 models need to be tested since the last deployment.

The thing is when it's on CI, the same command always pick up thousand of models (which is basically all of them). And I've used several ways to verify the manifests on remote are correct or not. And they're all good. I can use artifacts to cache all of them, and then download them to run those commands locally. I can see the numbers are the same which is 5 models.

It seems like this only happens when it's on CI.

I wonder if anyone experienced the same issue, or maybe have some clues?
Thanks for all the help in advance.

**Comment by jtcohen6** [comment]
This would likely benefit from the "explainable `state:modified`" proposed in:
- https://github.com/dbt-labs/dbt-core/issues/7109
- https://github.com/dbt-labs/dbt-core/pull/7102

**Reply by benbenbang** [reply]
Hello @jtcohen6, thanks for sharing.

If I understand correctly, this is currently an issue, and no clear explanation yet?

**Reply by jtcohen6** [reply]
Correct! Sorry, I could've been more clear. This is a proposed improvement to dbt, that would make it easier to debug scenarios like the one you've outlined above. I'm not exactly sure what's going on in your case :(

**Reply by benbenbang** [reply]
Alright, thank you, @jtcohen6. So I can only stay with the full models run before release at the moment 😿 
I would appreciate it if you could also suggest some suggestions on building the "slim CI" 🙏🏻
